### PR TITLE
Various pAI Tweaks + Sound Tweaks

### PR DESCRIPTION
**Pull request by Geevies**
* Mobs in bags can now hear sounds again.
* Various pAI bugfixes, such as the crew manifest and booping them not folding them up right.
* Disarming an unfolded pAI now boops them to get folded up.
* pAI have had their HP reduced a lot.

In the back-end, this changes sound to find visible turfs and then play the sound to all the mobs on the turf, instead of finding mobs directly.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/_helpers/game.dm [hunk 1]
```diff
@@ -184,6 +184,18 @@
...
+/proc/get_turfs_in_view(var/R, var/atom/source)
+	var/turf/start_point = get_turf(source)
+	var/list/hear = list()
+
+	if(!start_point)
+		return hear
+
+	var/list/range = hear(R, start_point)
+	for(var/turf/T in range)
+		hear += T
```
Comment (skull132) [inline code/_helpers/game.dm]: use an assoc list instead. `hear[T] = TRUE`

File: code/_helpers/game.dm [hunk 2]
```diff
@@ -184,6 +184,18 @@
...
 	return hear
 
+/proc/get_turfs_in_view(var/R, var/atom/source)
+	var/turf/start_point = get_turf(source)
+	var/list/hear = list()
+
+	if(!start_point)
+		return hear
+
+	var/list/range = hear(R, start_point)
```
Comment (skull132) [inline code/_helpers/game.dm]: `hear` is declared as a list in this function. What you're doing here is. Weird? You're not calling the function `hear()`, that's for sure.

File: code/game/sound.dm [hunk 3]
```diff
@@ -245,15 +245,14 @@ var/list/footstepfx = list("defaultstep","concretestep","grassstep","dirtstep","
...
+	var/list/turf/sound_turfs = get_turfs_in_view(world.view, source)
 	var/turf/source_turf = get_turf(source)
 
-	for (var/MM in mobs)
-		var/mob/M = MM
-		if (!M.sound_can_play(required_preferences, required_asfx_toggles))
+	for(var/mob/M in player_list)
+		if(!M.sound_can_play(required_preferences, required_asfx_toggles))
+			continue
+		if(!(get_turf(M) in sound_turfs))
```
Comment (skull132) [inline code/game/sound.dm]: Once you refactor to assoc list, you simply do `if (!sound_turfs[get_turf(M)])`

File: code/_helpers/game.dm [hunk 4]
```diff
@@ -184,6 +184,18 @@
...
 
+/proc/get_turfs_in_view(var/R, var/atom/source)
+	var/turf/start_point = get_turf(source)
+	var/list/hear_turfs = list()
+
+	if(!start_point)
+		return hear_turfs
+
+	var/list/range = hear(R, start_point)
+	for(var/turf/T in range)
```
Comment (skull132) [inline code/_helpers/game.dm]: This is technically a fuck-up. `view()` != `range()`. The proc's name is a lie.

File: code/modules/mob/living/silicon/pai/admin.dm [hunk 5]
```diff
@@ -5,26 +5,26 @@
...
 	if(!pai_key)
 		var/client/C = input("Select client") as null|anything in clients
-		if(!C) return
+		if(!C)
+			return
 		pai_key = C.key
 
-	log_and_message_admins("made a pAI with key=[pai_key] at ([t.x],[t.y],[t.z])")
-	var/obj/item/device/paicard/card = new(t)
+	log_and_message_admins("made a pAI with key=[pai_key] at ([T.x],[T.y],[T.z])")
```
Comment (skull132) [inline code/modules/mob/living/silicon/pai/admin.dm]: Coords unnecessary. They're added automatically based on `usr`.

Comment (skull132) [inline code/modules/mob/living/silicon/pai/admin.dm]:
```suggestion
	log_and_message_admins("made a pAI with key=[pai_key]")
```

**Comment by Geevies** [comment]
Closing due to lack of critical review.

**Pull request closed without merging**